samples/connectivity/bluetooth/companion/src/main/java/com/example/platform/connectivity/bluetooth/ble/CompanionDeviceManagerSample.kt

```diff
@@ -0,0 +1,304 @@
+                    Button(
+                        onClick = {
+                            onConnect(target)
+                        },
+                    ) {
+                        Text(text = "Connect")
+                    }
+                    Button(
+                        onClick = {
+                            scope.launch {
+                                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                                    deviceManager.disassociate(target.id)
+                                } else {
+                                    @Suppress("DEPRECATION")
+                                    deviceManager.disassociate(target.address)
+                                }
+                                associatedDevice = null
+                            }
+                        },
+                    ) {
+                        Text(text = "Disassociate")
+                    }
+                }
+            } else {
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(8.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                ) {
+                    Button(
+                        onClick = {
+                            scope.launch {
+                                val intentSender = requestDeviceAssociation(deviceManager)
+                                launcher.launch(IntentSenderRequest.Builder(intentSender).build())
+                            }
+                        },
+                    ) {
+                        Text(text = "Find & Associate device")
+                    }
+                    if (errorMessage.isNotBlank()) {
+                        Text(text = errorMessage, color = MaterialTheme.colorScheme.error)
+                    }
+                }
+            }
+        }
+    }
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+private fun getAssociatedDevice(deviceManager: CompanionDeviceManager): AssociatedDevice? {
+    val associatedDevice = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        deviceManager.myAssociations.lastOrNull()?.toAssociatedDevice()
+    } else {
+        // Before Android 34 we can only get the MAC. We could use the BT adapter to find the
+        // device, but to use CDM we only need the MAC.
+        @Suppress("DEPRECATION")
+        deviceManager.associations.lastOrNull()?.run {
+            AssociatedDevice(
+                id = -1,
+                address = this,
+                name = "",
+                device = null,
+            )
+        }
+    }
+    return associatedDevice
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+private fun Intent.getAssociationResult(): AssociatedDevice? {
+    var result: AssociatedDevice? = null
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        result = getParcelableExtra(
+            CompanionDeviceManager.EXTRA_ASSOCIATION,
```

Comment:
`EXTRA_ASSOCIATION` is only available for Android T+, which also deprecates `EXTRA_DEVICE`. You might want to use `EXTRA_DEVICE` here.